astroid/brain/brain_builtin_inference.py

```diff
@@ -1075,11 +1094,13 @@ def _infer_str_format_call(
 
 AstroidManager().register_transform(
     nodes.Call,
-    inference_tip(_infer_copy_method),
+    inference_tip(_infer_copy_method),  # type: ignore[arg-type]
```

Comment:
This can be avoided by using a `Generator` instead of an `Iterator` on the signature of this function.

Same goes for the other one, but that then creates other issues.. Can we yield from in that function? That would make it an iterator.

Reply:
Sorry, one more that I'm having trouble following :-)

Elsewhere we return iterators just fine, and they are typed this way:

https://github.com/pylint-dev/astroid/blob/e3ba1cadd4237d96d2e6c9a5341c7b652d389038/astroid/bases.py#L449-L453

So why would we change to Generator?

Reply:
Because `InferFn` expects a `Generator`. I think because we define it so explicitly we need to do so here as well.. I think we might also get away with changing `InferFn` to be an `Iterator` but since we don't actually run `mypy` that might create issues somewhere else that we don't know about..

Reply:
Ho! Got it now. Okay. I'll audit the uses of `InferFn`. It's start took like like it should be `Iterator`.

> but since we don't actually run mypy that might create issues somewhere else that we don't know about..

That's why I'm trying to focus by file and make sure with each change the total number goes down and no new issues in the single file I'm focusing on (but yes... whackamole could ensue...)

Reply:
Yeah, that does make the PRs much easier to review.

What I have also noticed is that doing PRs per concept (such as `InferFn`) is also a good way of grouping things. It might make the big number bigger as the improved typing exposes more issues, but it makes review and reasoning about the issues easier as it is all related.